packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -0,0 +1,198 @@
+  name: string;
+  parameters: Parameter[];
+  returnType: string;
+  isAuraEnabled: boolean;
+  className?: string;
+}
+
+/**
+ * Interface representing a parameter of a method.
+ */
+export interface Parameter {
+  name: string;
+  in: string;
+  required: boolean;
+  description: string;
+  schema: { type: string };
+}
+
+/**
+ * Class responsible for orchestrating metadata operations.
+ */
+export class MetadataOrchestrator {
+  /**
+   * Checks if a method is eligible for Apex Action creation.
+   * @param methodIdentifier - The identifier of the method.
+   * @returns True if the method is eligible, false otherwise.
+   */
+  public isMethodEligible = (methodIdentifier: string): boolean => {
+    // Placeholder for eligibility logic
+    return true;
+  };
+
+  /**
+   * Extracts metadata for the method at the current cursor position.
+   * @returns The metadata of the method, or undefined if no method is found.
+   */
+  public extractMethodMetadata = (): MethodMetadata | undefined => {
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
+      notificationService.showErrorMessage(nls.localize('no_active_editor'));
+      return;
+    }
+
+    const document = editor.document;
+    const cursorPosition = editor.selection.active;
+    const lines = document.getText().split('\n');
+    const currentLineIndex = cursorPosition.line;
+
+    let methodSignature = '';
+    let isAuraEnabled = false;
+
+    // Check if the preceding line contains @AuraEnabled
+    if (currentLineIndex > 0 && lines[currentLineIndex - 1].includes('@AuraEnabled')) {
+      isAuraEnabled = true;
+    }
+
+    // Traverse lines starting from the cursor position to construct the method signature
+    for (let line of lines) {
+      line = line.trim();
+      methodSignature += ` ${line}`;
+
+      // Stop once the closing parenthesis is reached
+      if (line.includes(') {')) {
+        break;
+      }
+    }
+
+    const methodMetadata = this.parseMethodSignature(methodSignature, isAuraEnabled);
+    if (!methodMetadata.isAuraEnabled) {
+      throw new Error(nls.localize('not_aura_enabled', methodMetadata.name));
+    }
+    return methodMetadata;
+  };
+
+  public extractAllMethodsMetadata = (): MethodMetadata[] | undefined => {
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
+      notificationService.showErrorMessage(nls.localize('no_active_editor'));
+      return;
+    }
+
+    const document = editor.document;
+    const filePath = document.fileName;
+    const className = filePath
+      .substring(filePath.lastIndexOf(process.platform === 'win32' ? '\\' : '/') + 1)
+      .split('.')
+      .shift();
+    const lines = document.getText().split('\n');
+    const metadataList: MethodMetadata[] = [];
+    let currentMethodSignature = '';
+    let isAuraEnabled = false;
+
+    for (let line of lines) {
+      line = line.trim();
+
+      // Detect @AuraEnabled annotation
+      if (line.includes('@AuraEnabled')) {
```

Comment:
I think there is no need to capture the method metadata here. What we need is the file URI and location (if available), and the language server will return the method document symbol with the eligibility to the client side.

Reply:
Yes, but we still don't have any of that logic so I thought it'd be nice to keep this one until then to be able to demo if needed